sdk/communication/AzureCommunicationChat/Source/Signaling/events/chat.swift

```diff
@@ -0,0 +1,676 @@
+    public var removedOn: Iso8601Date?
+    public var participantsRemoved: [SignalingChatParticipant]?
+    public var removedBy: SignalingChatParticipant?
+
+    init(
+        threadId: String,
+        version: String,
+        removedOn: Iso8601Date?,
+        participantsRemoved: [SignalingChatParticipant]?,
+        removedBy: SignalingChatParticipant?
+    ) {
+        self.removedOn = removedOn
+        self.participantsRemoved = participantsRemoved
+        self.removedBy = removedBy
+        super.init(threadId: threadId, version: version)
+    }
+
+    init(from request: TrouterRequest) throws {
+        guard let requestJsonData = request.body.data(using: .utf8) else {
+            throw AzureError.client("Unable to convert request body to Data.")
+        }
+
+        let participantsRemovedPayload: ParticipantsRemovedPayload = try JSONDecoder()
+            .decode(ParticipantsRemovedPayload.self, from: requestJsonData)
+
+        guard let removedByJsonData = participantsRemovedPayload.removedBy.data(using: .utf8) else {
+            throw AzureError.client("Unable to convert payload removedBy to Data.")
+        }
+
+        let removedByPayload: ChatParticipantPayload = try JSONDecoder()
+            .decode(ChatParticipantPayload.self, from: removedByJsonData)
+        let removedBy = SignalingChatParticipant(
+            id: TrouterEventUtil.getIdentifier(from: removedByPayload.participantId),
+            displayName: removedByPayload.displayName
+        )
+
+        guard let participantsJsonData = participantsRemovedPayload.participantsRemoved.data(using: .utf8) else {
+            throw AzureError.client("Unable to convert payload participantsRemoved to Data.")
+        }
+
+        let participantsPayload: [ChatParticipantPayload] = try JSONDecoder()
+            .decode([ChatParticipantPayload].self, from: participantsJsonData)
+        let participants: [SignalingChatParticipant] = participantsPayload
+            .map { (memberPayload: ChatParticipantPayload) -> SignalingChatParticipant in
+                SignalingChatParticipant(
+                    id: TrouterEventUtil.getIdentifier(from: memberPayload.participantId),
+                    displayName: memberPayload.displayName,
+                    shareHistoryTime: Iso8601Date(
+                        string: TrouterEventUtil
+                            .toISO8601Date(unixTime: memberPayload.shareHistoryTime)
+                    )
+                )
+            }
+
+        self.removedOn = Iso8601Date(string: participantsRemovedPayload.time)
+        self.participantsRemoved = participants
+        self.removedBy = removedBy
+        super.init(
+            threadId: participantsRemovedPayload.threadId,
+            version: participantsRemovedPayload.version
+        )
+    }
+}
+
+public enum ChatEventId: String {
+    case chatMessageReceived
+    case typingIndicatorReceived
+    case readReceiptReceived
```

Comment:
This might be the means by which you can get rid of the `Any` return type. We can explore prior to GA.

Reply:
Issue I ran into is there isn't one base class that chat events inherit from, I also notice some classes that look very similar to existing models in Chat so maybe we can get rid of them entirely. Prior to GA I think it's good to review all the classes in `chat.swift`